src/custom/hooks/useMetadataDoc.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from 'react'
+import { useActiveWeb3React } from 'hooks/web3'
+
+interface AffiliateDoc {
+  version: string
+  appCode: string
+  metadata: any
+}
+
+export function useMetadataDoc() {
+  const { account } = useActiveWeb3React()
+  const [document, setDocument] = useState(<AffiliateDoc>{
+    version: '1.0.0',
+    appCode: 'CowSwap',
+    metadata: {},
+  })
```

Comment:
In the metadata.ts file there's already a metadata type defined including a method to create it.